File: LeetCodeRecord/MethodFirst/240-SearchMatrix.java
package  MethodFirst;

class SearchMatrix {
    public boolean searchMatrix(int[][] matrix, int target) {
        int rows = matrix.length;
        if (rows == 0) {
            return false;
        }
        int columns = matrix[0].length;

        int rowNum = 0;
        int colNum = columns - 1;
        while (rowNum < rows && colNum >= 0) {
            if (target > matrix[rowNum][colNum]) {
                rowNum ++;
            } else if (target < matrix[rowNum][colNum]) {
                colNum --;
            } else {
                return true;
            }
        }

        return false;
    }

    public static void main(String[] args) {
    
    }
}

File: LeetCodeRecord/ACFirst-Golang/240-SearchMatrix.go
package main

func searchMatrix(matrix [][]int, target int) bool {
	rows := len(matrix)
	if rows == 0 {
		return false
	}
	columns := len(matrix[0])

	rowNum := 0
	columnNum := columns - 1

	for rowNum < rows && columnNum >= 0 {
		if target > matrix[rowNum][columnNum] {
			rowNum++
		} else if target < matrix[rowNum][columnNum] {
			columnNum--
		} else {
			return true
		}
	}

	return false
}
